spec/requests/settings_spec.rb

```diff
@@ -0,0 +1,115 @@
+describe 'Settings API' do
+  let!(:tenant) do
+    create(:tenant,
+           :settings => {
+             :icon             => "<svg rel='stylesheet'>image</svg>",
+             :default_workflow => "1"
+           })
+  end
+  let(:retreived_tenant) { Tenant.find(tenant.id) }
+
+  let(:org_admin_hash) do
+    user = default_user_hash
+    user["identity"]["user"]["is_org_admin"] = true
+    encoded_user_hash(user)
+  end
+
+  let(:manipulated_headers) do
+    { 'x-rh-identity'            => org_admin_hash,
+      'x-rh-insights-request-id' => 'gobbledygook' }
+  end
+
+  context "when the user is an org admin" do
+    describe "#index" do
+      before { get "#{api}/settings", :headers => manipulated_headers }
+
+      it "returns the current settings of the tenant" do
+        expect(response).to have_http_status(:ok)
+        expect(json["current"]["icon"]).to eq retreived_tenant.icon
+        expect(json["current"]["default_workflow"]).to eq retreived_tenant.default_workflow
+      end
+
+      it 'returns the json schema as well' do
+        expect(json["schema"]).to eq JSON.parse(Catalog::TenantSettings.new(tenant).send(:schema))
+      end
+    end
+
+    describe "#show" do
+      before { get "#{api}/settings/icon", :headers => manipulated_headers }
+
+      it "returns the specified setting" do
+        expect(response).to have_http_status(:ok)
+        expect(json["icon"]).to eq retreived_tenant.icon
+      end
+    end
+
+    context "when the key exists" do
```

Comment:
'when the key exists` and `when the key already exists` ( line 77 ) are very close. Possibly move the context on line 77 into this one?

Reply:
Ok reworked the specs so they're a bit easier to read.